src/map/packets/char_stats.cpp

```diff
@@ -57,7 +57,7 @@ CCharStatsPacket::CCharStatsPacket(CCharEntity* PChar)
     ref<uint16>(0x2C) = std::clamp<int16>(PChar->getMod(Mod::MND), -999 + PChar->stats.MND, 999 - PChar->stats.MND);
     ref<uint16>(0x2E) = std::clamp<int16>(PChar->getMod(Mod::CHR), -999 + PChar->stats.CHR, 999 - PChar->stats.CHR);
 
-    ref<uint16>(0x30) = PChar->ATT();
+    ref<uint16>(0x30) = PChar->ATT(SLOT_MAIN);
```

Comment:
See comment ref: the default values.  I don't believe this is the cause of the issue, but more likely slot being out of range.

Reply:
0x11 was originally done to remove it from the equip screen. I think we should go back and look at where ATT() is referenced or we can inside of ATT() put in a bit for if it is not SLOT_MAIN -> SLOT_AMMO to ignore weapons.

Reply:
Ah. I understand now. I actually missed that it has a default value assigned.
This packet is used in things other than equip screen though (which is why crashes could happen).

Reply:
@OpheliaXI my concern with saying "if this slot is not SLOT_MAIN -> SLOT_AMMO" would be that we may not catch unexpected usage of the method. I.e: If someone were to call `PChar->ATT(SLOT_BODY)` and they intended to call it with a valid weapon slot, the code would just run fine without complaining. It is annoying to add an extra parameter but I believe it's best to separate the error case from the valid input.

Reply:
Oh I see now from disc that we agreed on ATT(slot, bool ignoreWeapon = false);. That makes sense to me.

Reply:
Actually reading this again, if we ignore weapon att from every CCharStats packet, won't that show invalid att in the equip screen?

Reply:
A bool would solve this problem as the equip screen should show no att values from weapons.

Reply:
I implemented the bool to stop this from going mad and reverted the overload for the uint16 to SLOT_MAIN. As for the bool, it is overloaded to false to allow for backwards compatibility with other functions.